--- ceph-10.2.11/src/test/librados_test_stub/TestWatchNotify.cc.orig	2018-07-09 18:23:56.000000000 +0200
+++ ceph-10.2.11/src/test/librados_test_stub/TestWatchNotify.cc	2020-05-23 14:37:53.766463510 +0200
@@ -1,6 +1,7 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 
+#define BOOST_DISABLE_ASSERTS // broken with ceph assert.h
 #include "test/librados_test_stub/TestWatchNotify.h"
 #include "include/Context.h"
 #include "common/Finisher.h"
