--- ceph-12.2.13/cmake/modules/Findgperftools.cmake.orig	2020-01-30 21:52:36.000000000 +0100
+++ ceph-12.2.13/cmake/modules/Findgperftools.cmake	2020-05-28 17:05:49.948032294 +0200
@@ -11,13 +11,13 @@
 find_path(GPERFTOOLS_INCLUDE_DIR gperftools/profiler.h)
 find_path(Tcmalloc_INCLUDE_DIR gperftools/tcmalloc.h)
 
-foreach(component tcmalloc tcmalloc_minimal profiler)
+foreach(component tcmalloc tcmalloc_minimal)
   string(TOUPPER ${component} COMPONENT)
   find_library(GPERFTOOLS_${COMPONENT}_LIBRARY ${component})
   list(APPEND GPERFTOOLS_LIBRARIES GPERFTOOLS_${COMPONENT}_LIBRARY)
 endforeach()
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(gperftools DEFAULT_MSG GPERFTOOLS_LIBRARIES GPERFTOOLS_INCLUDE_DIR)
+find_package_handle_standard_args(gperftools DEFAULT_MSG GPERFTOOLS_LIBRARIES Tcmalloc_INCLUDE_DIR)
 
-mark_as_advanced(GPERFTOOLS_LIBRARIES GPERFTOOLS_INCLUDE_DIR)
+mark_as_advanced(GPERFTOOLS_LIBRARIES GPERFTOOLS_INCLUDE_DIR Tcmalloc_INCLUDE_DIR)
