diff -up ceph-0.41/src/init-ceph.in.init ceph-0.41/src/init-ceph.in
--- ceph-0.41/src/init-ceph.in.init	2012-01-17 15:33:20.000000000 -0500
+++ ceph-0.41/src/init-ceph.in	2012-02-16 12:48:50.887279921 -0500
@@ -1,11 +1,11 @@
 #!/bin/sh
 # Start/stop ceph daemons
-# chkconfig: 2345 60 80
+# chkconfig: - 60 80
 
 ### BEGIN INIT INFO
 # Provides:          ceph
-# Default-Start:     2 3 5
-# Default-Stop:      0 1 6
+# Default-Start:     
+# Default-Stop:      
 # Required-Start:    $remote_fs $named $network $time
 # Required-Stop:     $remote_fs $named $network $time
 # Short-Description: Start Ceph distributed file system daemons at boot time
