--- ceph-16.2.7/cmake/modules/Distutils.cmake~	2021-12-07 17:15:49.000000000 +0100
+++ ceph-16.2.7/cmake/modules/Distutils.cmake	2022-02-12 11:15:17.620439748 +0100
@@ -19,12 +19,10 @@
   install(CODE "
     set(options --prefix=${CMAKE_INSTALL_PREFIX})
     if(DEFINED ENV{DESTDIR})
-      if(EXISTS /etc/debian_version)
-        list(APPEND options --install-layout=deb)
-      endif()
       list(APPEND options
         --root=\$ENV{DESTDIR}
-        --single-version-externally-managed)
+        --single-version-externally-managed
+	--optimize=2)
     endif()
     if(NOT \"${DU_INSTALL_SCRIPT}\" STREQUAL \"\")
       list(APPEND options --install-script=${DU_INSTALL_SCRIPT})
@@ -109,10 +107,7 @@
 
     set(options --prefix=${CMAKE_INSTALL_PREFIX})
     if(DEFINED ENV{DESTDIR})
-      if(EXISTS /etc/debian_version)
-        list(APPEND options --install-layout=deb)
-      endif()
-      list(APPEND options --root=\$ENV{DESTDIR})
+      list(APPEND options --root=\$ENV{DESTDIR} --optimize=2)
     else()
       list(APPEND options --root=/)
     endif()
--- ceph-18.2.3/src/cephadm/build.py.orig	2024-04-24 21:57:21.000000000 +0200
+++ ceph-18.2.3/src/cephadm/build.py	2024-05-30 20:51:47.139755410 +0200
@@ -85,7 +85,7 @@ def _compile(dest, tempdir):
         maxlevels=16,
         legacy=True,
         quiet=1,
-        workers=0,
+        workers=1,
     )
     # TODO we could explicitly pass a python version here
     log.info("Constructing the zipapp file")
