--- ceph-0.94.1/configure.ac.orig	2015-04-10 21:09:16.000000000 +0200
+++ ceph-0.94.1/configure.ac	2015-04-23 17:09:49.415097906 +0200
@@ -967,7 +967,7 @@
 ])
 
 AC_MSG_CHECKING([for sched.h])
-+AC_LANG_PUSH([C++])
+AC_LANG_PUSH([C++])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #define _GNU_SOURCE
 #include <sched.h>
@@ -984,7 +984,7 @@
 ], [
 AC_MSG_RESULT([no])
 ])
-+AC_LANG_POP([C++])
+AC_LANG_POP([C++])
 
 
 #
