--- ceph-0.55.1/src/logrotate.conf.orig	2012-12-15 15:11:13.516032449 +0100
+++ ceph-0.55.1/src/logrotate.conf	2012-12-15 15:16:52.636025383 +0100
@@ -4,23 +4,7 @@
     compress
     sharedscripts
     postrotate
-        if which invoke-rc.d && [ -x `which invoke-rc.d` ]; then
-            invoke-rc.d ceph reload >/dev/null
-        elif which service && [ -x `which service` ]; then
-            service ceph reload >/dev/null
-        fi
-        # Possibly reload twice, but depending on ceph.conf the reload above may be a no-op
-        if which initctl && [ -x `which initctl` ]; then
-            # upstart reload isn't very helpful here:
-            #   https://bugs.launchpad.net/upstart/+bug/1012938
-            for type in mon osd mds; do
-		initctl list \
-		    | perl -ne 'print "$+{service} cluster=$+{cluster} id=$+{id}\n" if m{^(?<service>ceph-(mon|osd|mds)+)\s+\((?<cluster>[^/)]+)/(?<id>[^)]+)\) start/}' \
-		    | while read l; do
-		    initctl reload -- $l 2>/dev/null || :
-		done
-            done
-        fi
+        service ceph reload >/dev/null 2>/dev/null
     endscript
     missingok
 }
