--- ceph-15.2.16/src/test/fio/CMakeLists.txt.orig	2022-06-12 19:33:32.012668509 +0200
+++ ceph-15.2.16/src/test/fio/CMakeLists.txt	2022-06-12 19:35:24.095394638 +0200
@@ -27,7 +27,7 @@ if(WITH_FIO)
 endif()
 
 target_link_libraries(fio_ceph_objectstore os global)
-install(TARGETS fio_ceph_objectstore DESTINATION lib)
+install(TARGETS fio_ceph_objectstore DESTINATION lib${LIB_SUFFIX})
 
 target_link_libraries(fio_ceph_messenger os global)
-install(TARGETS fio_ceph_messenger DESTINATION lib)
+install(TARGETS fio_ceph_messenger DESTINATION lib${LIB_SUFFIX})
