diff -up ceph-0.41/src/cephfs.cc.compilefix ceph-0.41/src/cephfs.cc
--- ceph-0.41/src/cephfs.cc.compilefix	2012-02-16 21:08:19.983652818 -0500
+++ ceph-0.41/src/cephfs.cc	2012-02-16 21:08:31.917550083 -0500
@@ -24,6 +24,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <stdlib.h>
+#include <unistd.h>
 #include "client/ioctl.h"
 
 using namespace std;
diff -up ceph-0.41/src/common/safe_io.h.compilefix ceph-0.41/src/common/safe_io.h
--- ceph-0.41/src/common/safe_io.h.compilefix	2012-02-16 12:53:31.456514774 -0500
+++ ceph-0.41/src/common/safe_io.h	2012-02-16 12:54:00.045233017 -0500
@@ -16,6 +16,7 @@
 #define CEPH_SAFE_IO
 
 #include "common/compiler_extensions.h"
+#include <sys/types.h>
 
 #ifdef __cplusplus
 extern "C" {
diff -up ceph-0.41/src/osd/PG.cc.compilefix ceph-0.41/src/osd/PG.cc
--- ceph-0.41/src/osd/PG.cc.compilefix	2012-02-16 20:58:21.039559849 -0500
+++ ceph-0.41/src/osd/PG.cc	2012-02-16 20:59:07.920149937 -0500
@@ -4830,13 +4830,13 @@ bool PG::PriorSet::affected_by_map(const
     }
 
     // did a down osd in cur get (re)marked as lost?
-    map<int,epoch_t>::const_iterator p = blocked_by.find(o);
-    if (p != blocked_by.end()) {
+    map<int,epoch_t>::const_iterator r = blocked_by.find(o);
+    if (r != blocked_by.end()) {
       if (!osdmap->exists(o)) {
 	dout(10) << "affected_by_map osd." << o << " no longer exists" << dendl;
 	return true;
       }
-      if (osdmap->get_info(o).lost_at != p->second) {
+      if (osdmap->get_info(o).lost_at != r->second) {
 	dout(10) << "affected_by_map osd." << o << " (re)marked as lost" << dendl;
 	return true;
       }
diff -up ceph-0.41/src/osd/ReplicatedPG.cc.compilefix ceph-0.41/src/osd/ReplicatedPG.cc
--- ceph-0.41/src/osd/ReplicatedPG.cc.compilefix	2012-02-16 21:03:26.140892128 -0500
+++ ceph-0.41/src/osd/ReplicatedPG.cc	2012-02-16 21:04:45.190200945 -0500
@@ -5436,8 +5436,8 @@ int ReplicatedPG::recover_replicas(int m
       }
 
       dout(10) << __func__ << ": recover_object_replicas(" << soid << ")" << dendl;
-      map<hobject_t,Missing::item>::const_iterator p = m.missing.find(soid);
-      started += recover_object_replicas(soid, p->second.need);
+      map<hobject_t,Missing::item>::const_iterator r = m.missing.find(soid);
+      started += recover_object_replicas(soid, r->second.need);
     }
   }
 
diff -up ceph-0.41/src/test/system/systest_runnable.cc.compilefix ceph-0.41/src/test/system/systest_runnable.cc
--- ceph-0.41/src/test/system/systest_runnable.cc.compilefix	2012-02-16 12:54:35.430884275 -0500
+++ ceph-0.41/src/test/system/systest_runnable.cc	2012-02-16 12:54:45.923780862 -0500
@@ -28,6 +28,7 @@
 #include <sys/syscall.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <unistd.h>
 #include <vector>
 
 #if defined(__FreeBSD__)
