--- ceph-12.2.13/cmake/modules/Distutils.cmake.orig	2020-01-30 21:52:36.000000000 +0100
+++ ceph-12.2.13/cmake/modules/Distutils.cmake	2020-05-29 22:15:43.919231143 +0200
@@ -17,10 +17,7 @@
   install(CODE "
     set(options --prefix=${CMAKE_INSTALL_PREFIX})
     if(DEFINED ENV{DESTDIR})
-      if(EXISTS /etc/debian_version)
-        list(APPEND options --install-layout=deb)
-      endif()
-      list(APPEND options --root=\$ENV{DESTDIR})
+      list(APPEND options --root=\$ENV{DESTDIR} --optimize=2)
       if(NOT \"${DU_INSTALL_SCRIPT}\" STREQUAL \"\")
         list(APPEND options --install-script=${DU_INSTALL_SCRIPT})
       endif()
