--- ceph-12.2.13/src/logrotate.conf.orig	2020-05-28 06:45:00.496496556 +0200
+++ ceph-12.2.13/src/logrotate.conf	2020-05-28 06:47:07.249143211 +0200
@@ -4,8 +4,7 @@
     compress
     sharedscripts
     postrotate
-        killall -q -1 ceph-mon ceph-mgr ceph-mds ceph-osd ceph-fuse radosgw || true
-        killall -q -1 ceph-mon ceph-mgr ceph-mds ceph-osd ceph-fuse radosgw rbd-mirror || true
+        cephctl reload >/dev/null 2>/dev/null
     endscript
     missingok
     notifempty
