--- ceph-16.2.7/cmake/modules/CheckNasm.cmake~	2021-12-07 17:15:49.000000000 +0100
+++ ceph-16.2.7/cmake/modules/CheckNasm.cmake	2022-02-12 19:19:51.463245367 +0100
@@ -10,6 +10,9 @@
       set(CMAKE_REQUIRED_QUIET true)
       include(CheckCXXSourceCompiles)
       check_cxx_source_compiles("
+      #if !defined(__x86_64__)
+      #error ix86
+      #endif
       #if defined(__x86_64__) && defined(__ILP32__)
       #error x32
       #endif
