--- ceph-19.2.0/src/cephadm.orig/build.py	2024-12-22 14:34:36.723333480 +0100
+++ ceph-19.2.0/src/cephadm/build.py	2024-12-22 15:09:43.746666771 +0100
@@ -51,12 +51,12 @@
 PY36_REQUIREMENTS = [
     {
         'package_spec': 'MarkupSafe >= 2.0.1, <2.2',
-        'from_source': True,
+        'from_source': False,
         'unique': True,
     },
     {
         'package_spec': 'Jinja2 >= 3.0.2, <3.2',
-        'from_source': True,
+        'from_source': False,
         'unique': True,
     },
     {
@@ -69,8 +69,8 @@
     },
 ]
 PY_REQUIREMENTS = [
-    {'package_spec': 'MarkupSafe >= 2.1.3, <2.2', 'from_source': True},
-    {'package_spec': 'Jinja2 >= 3.1.2, <3.2', 'from_source': True},
+    {'package_spec': 'MarkupSafe >= 2.1.3, <2.2', 'from_source': False},
+    {'package_spec': 'Jinja2 >= 3.1.2, <3.2', 'from_source': False},
     # We can not install PyYAML using sources. Unlike MarkupSafe it requires
     # Cython to build and Cython must be compiled and there's not clear way past
     # the requirement in pyyaml's pyproject.toml. Instead, rely on fetching
@@ -161,7 +161,7 @@
     def __init__(self, cli_args):
         self.cli_args = cli_args
         self._maj_min = sys.version_info[0:2]
-        self.install_dependencies = True
+        self.install_dependencies = False
         self.deps_mode = DependencyMode[cli_args.bundled_dependencies]
         if self.deps_mode == DependencyMode.none:
             self.install_dependencies = False
